src/ex1/script.js

```diff
@@ -0,0 +1,284 @@
+todoElement.addEventListener("click", alertTask);
+
+//Functions
+
+//add alert feature when task is clicked
+function alertTask(e) {
+  const item = e.target;
+
+  if (item.classList[0] === "todo") {
+    const todo = item.parentElement;
+
+    let message = "📝" + item.innerText;
+    return alert(item.innerText);
+    /*will be used later in the project launch_listElement(message); */
+  }
+}
+
+//deactive / active buttons when todos are added / cleared
+function listStyling(todosCount) {
+  if (!todosCount) {
+    clearAllBtn.classList.add("deactive");
+    sortBtn.classList.add("deactive");
+    footerSpan.classList.add("deactive");
+  } else {
+    clearAllBtn.classList.remove("deactive");
+    sortBtn.classList.remove("deactive");
+    footerSpan.classList.remove("deactive");
+  }
+}
+
+//todo input styling
+todoInput.onkeydown = () => {
+  if (todoInput.value.length) {
+    inputBox.classList.add("inactive");
+    todoButton.classList.add("inactive");
+  }
+
+  if (!todoInput.value.length && !todoList.childElementCount) {
+    inputBox.classList.remove("inactive");
+    todoButton.classList.remove("inactive");
+  }
+};
+
+//add todo to the todo list
+function addTodo(event) {
+  if (!todoInput.value.trim()) {
+    return launch_toast();
+  }
+  event.preventDefault();
+
+  //Todo li
+  animItem.goToAndPlay(0, true);
+  const todoLi = document.createElement("li");
+  todoLi.classList.add("todo");
+  todoLi.innerText = todoInput.value;
+
+  //Add todo to local storage
+  saveLocalTodos(todoInput.value);
+
+  //trash button
+  const trashButton = document.createElement("button");
+  trashButton.innerHTML = '<i  class="fas fa-trash"></i>';
+  trashButton.classList.add("trash-btn");
+  todoLi.appendChild(trashButton);
+  //append to list
+  todoList.appendChild(todoLi);
+
+  const listChildrenCount = todoList.childElementCount;
+  listStyling(listChildrenCount);
+
+  //Clear Todo Input value
+  todoInput.value = "";
+  inputBox.classList.add("inactive");
+}
+
+//add todo when pressing enter key
+function todoEnter(event) {
+  if (event.keyCode == ENTER_KEY) {
+    if (!todoInput.value.trim()) {
+      return launch_toast();
+    }
+    addTodo(event);
+  }
+}
+
+//delete a todo from the list
+function deleteTask(e) {
+  const item = e.target;
+
+  if (item.classList[0] === "trash-btn") {
+    const todo = item.parentElement;
+    todo.classList.add("fall");
+    removeLocalTodos(todo);
+    //delay the remove for animation
+    setTimeout(function () {
+      todo.remove();
+    }, TRASH_ANIMATION_TIMEOUT);
+  }
+}
+
+//clear all button clicked
+function clearAll(e) {
+  const item = e.target;
+  if (item.classList[0] === "clearAllBtn") {
+    todoList.classList.add("fall");
+
+     //delay the remove for animation
+    setTimeout(function () {
+      todoList.innerHTML = "";
+      localStorage.clear();
+      todoList.classList.remove("fall");
+      listStyling(0);
+    }, TRASH_ANIMATION_TIMEOUT);
```

Comment:
It is usually a bad practice to hold of business logic because of UI / animations.
They should not be coupled with each other.